readthedocs/core/middleware.py

```diff
@@ -12,52 +12,71 @@
 log = logging.getLogger(__name__)
 
 LOG_TEMPLATE = u"(Middleware) {msg} [{host}{path}]"
+SUBDOMAIN_URLCONF = getattr(
+    settings,
+    'SUBDOMAIN_URLCONF',
+    'readthedocs.core.subdomain_urls'
+)
+SINGLE_VERSION_URLCONF = getattr(
+    settings,
+    'SINGLE_VERSION_URLCONF',
+    'readthedocs.core.single_version_urls'
+)
 
 
 class SubdomainMiddleware(object):
 
     def process_request(self, request):
+        if not getattr(settings, 'USE_SUBDOMAIN', False):
+            return None
+
         host = request.get_host().lower()
         path = request.get_full_path()
         log_kwargs = dict(host=host, path=path)
-        if settings.DEBUG:
-            log.debug(LOG_TEMPLATE.format(msg='DEBUG on, not processing middleware', **log_kwargs))
-            return None
+        public_domain = getattr(settings, 'PUBLIC_DOMAIN', None)
+        production_domain = settings.PRODUCTION_DOMAIN
```

Comment:
Only because it's not an optional setting like `PUBLIC_DOMAIN`. It could be set that way, however. In fact, maybe it makes sense to move the many instances of `getattr(settings, 'PRODUCTION_DOMAIN')` to `core.constants` or something.